inbm/dispatcher-agent/dispatcher/update_logger.py

```diff
@@ -187,6 +201,25 @@ def update_granular_with_sota_package_list_install(self) -> None:
             with open(GRANULAR_LOG_FILE, 'w') as f:
                 json.dump(data, f, indent=4)
 
+    def update_granular_with_log(self, log: dict) -> None:
+        """This function stores the log provided into the granular log file."""
+        logger.debug("")
+        try:
+            self._granular_lock.acquire()
```

Comment:
instead of

```python
try:
    lock.acquire()
    # code
finally:
    lock.release()
```

do this:


```python
with lock:
   # code
```